File: csharp/DVSClient/DVSClient/address/lookup/LookupAddressSuggestionV2.cs
using DVSClient.Server.Address.Lookup;

namespace DVSClient.Address.Lookup
{
    public class LookupAddressSuggestionV2
    {
        public string? Text { get; }
        public IEnumerable<int>? Matched { get; }
        public string? GlobalAddressKey { get; }
        public string? Format { get; }
        public string? Dataset { get; }
        public IEnumerable<string>? Names { get; }
        public string? Uprn { get; }

        public LookupAddressSuggestionV2(RestApiAddressLookupSuggestionV2 response)
        {
            Text = response.Text;
            Matched = response.Matched;
            GlobalAddressKey = response.GlobalAddressKey;
            Format = response.Format;
            Dataset = response.Dataset;
            Names = response.Names;
            Uprn = response.Uprn;
        }
    }
}

File: typescript/src/address/lookup/lookupAddressSuggestionV2.ts
import { RestApiAddressLookupSuggestionV2 } from "../../server/address/lookup/restApiAddressLookupSuggestionV2";

export type LookupAddressSuggestionV2 = {
    text?: string;
    matched?: number[];
    globalAddressKey?: string;
    format?: string;
    dataset?: string;
    names?: string[];
    uprn?: string;
}

export function restApiResponseToLookupAddressSuggestionV2Hub(response: RestApiAddressLookupSuggestionV2): LookupAddressSuggestionV2 {
    const result: LookupAddressSuggestionV2 = {
        text: response.text,
        matched: response.matched,
        globalAddressKey: response.global_address_key,
        format: response.format,
        dataset: response.dataset,
        names: response.names,
        uprn: response.uprn
    }
    return result;
}